core/src/main/scala/org/apache/spark/launcher/LauncherBackend.scala

```diff
@@ -71,6 +110,9 @@ private[spark] abstract class LauncherBackend {
     if (connection != null && lastState != state) {
       connection.send(new SetState(state))
       lastState = state
+      if(!_isConnected && stopFlag) {
```

Comment:
so when does this happen?  We are in code that the connection isn't null and we are changing the state but we aren't connected (ie close() has been called)?